mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp

```diff
@@ -199,6 +200,91 @@ bool mlir::sycl::SYCLCastOp::areCastCompatible(::mlir::TypeRange Inputs,
   return HasArrayTrait && IsArray;
 }
 
+mlir::LogicalResult mlir::sycl::SYCLGEPOp::verify() {
+  // /* Available only when: (Dimensions > 0) */
+  // reference operator[](id<Dimensions> index) const;
+
+  // /* Available only when: (Dimensions > 1) */
+  // __unspecified__ operator[](size_t index) const;
+
+  // /* Available only when: (AccessMode != access_mode::atomic && Dimensions ==
+  // 1) */
+  // reference operator[](size_t index) const;
+  const auto AccessorTy = getOperand(0)
+                              .getType()
+                              .cast<mlir::MemRefType>()
```

Comment:
Worth to assert that operant 0 is a MemRefType ?

Reply:
It will always be as the operation wouldn't be created otherwise,.